lib/ansible/modules/cloud/amazon/ec2_spot_pricing_history_facts.py

```diff
@@ -0,0 +1,225 @@
+        retrieving the price history data.
+    default: ansible_date_time.iso8601
+
+  end_time:
+    description:
+      - >
+        The date and time, up to the current date, from which to stop retrieving
+        the price history data.
+    default: ansible_date_time.iso8601
+
+  max_results:
+    description:
+      - the total number of items to return in the output.
+    default: 1000
+
+  filters:
+    description:
+      - One or more filters
+    default: []
+
+extends_documentation_fragment:
+    - aws
+    - ec2
+"""
+
+EXAMPLES = """
+# Find the most recent spot price for t3.medium in eu-west-1a
+- ec2_spot_price_history_facts:
+    instance_types: ["t3.medium"]
+    availability_zone: "eu-west-1a"
+    product_descriptions: ["Linux/UNIX"]
+    max_results: 1
+  register: spot_prices
+
+# Create an instance using the spot price
+- ec2:
+   spot_price: "{{ spot_pricing_history[0].spot_price }}"
+   vpc_subnet_id: "subnet-07b18fbad727f2444"
+   image: "ami-00035f41c82244dab"
+   instance_type: "t3.medium"
+   group: "my-security-group"
+   wait: true
+   instance_tags:
+     Name: "my spot instance"
+"""
+
+RETURN = """
+---
+ec2_spot_pricing_history_facts:
+  description: The Amazon Resource Name of the ASG
+  returned: success
+  type: list
+  sample: [
+    {
+      "availability_zone": "eu-west-1a",
+      "spot_price": 0.71,
+      "instance_type": "t3.medium",
+      "product_description": "Linux/UNIX",
+      "timestamp": "2019-02-12T13:47:13+00:00"
+    }
+  ]
+"""
+
+import datetime
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict,
+)
+
+try:
+    import dateutil.parser
+
+    HAS_DATEUTIL = True
+except ImportError:
+    HAS_DATEUTIL = False
+
+
+def parse_date(module, key):
+    value = module.params[key]
+    try:
+        start_time = dateutil.parser.parse(value)
+        return start_time
+    except ValueError:
+        module.fail_json(
+            msg="{key}={value} is not a valid date".format(key=key, value=value)
+        )
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            availability_zone=dict(type="str", default=""),
+            product_descriptions=dict(
+                type="list",
+                default=[],
+                choices=[
+                    "Linux/UNIX",
+                    "SUSE Linux",
+                    "Windows",
+                    "Linux/UNIX (Amazon VPC)",
+                    "SUSE Linux (Amazon VPC)",
+                    "Windows (Amazon VPC)",
+                ],
+            ),
+            instance_types=dict(type="list", default=[]),
+            filters=dict(type="list", default=[]),
+            start_time=dict(
+                type="str",
+                default=datetime.datetime.now().replace(microsecond=0).isoformat(),
```

Comment:
Defaults need to be static values. Values of `None` or `"now"` could be used to indicate the current time.